File: Go/pkg/sort/insertion_sort.go
package sort

func insertionSort(valuesToSort []int) []int {
	n := len(valuesToSort)
	internalValueToSort := make([]int, n)
	copy(internalValueToSort, valuesToSort)
	for i := 1; i < n; i++ {
		j := i - 1
		value := internalValueToSort[i]
		for j >= 0 && internalValueToSort[j] > value {
			internalValueToSort[j+1] = internalValueToSort[j]
			j = j - 1
		}
		internalValueToSort[j+1] = value
	}
	return internalValueToSort
}

File: Java/sorting/algoritms/src/main/java/org/example/InsertionSort.java
package org.example;

import java.util.ArrayList;
import java.util.List;

public class InsertionSort {
    public List<Integer> sort(List<Integer> valuesToSort) {
        int n = valuesToSort.size();
        List<Integer> internalValuesToSort = new ArrayList<>(valuesToSort);
        for (int i = 1; i < n; i++) {
            int j = i - 1;
            int value = internalValuesToSort.get(i);
            while (j >= 0 && internalValuesToSort.get(j) > value) {
                internalValuesToSort.set(j + 1, internalValuesToSort.get(j));
                j--;
            }
            internalValuesToSort.set(j + 1, value);
        }
        return internalValuesToSort;
    }
}
